ReactWindows/ReactNative/Views/DatePicker/ReactCalendarDatePickerManager.cs

```diff
@@ -0,0 +1,232 @@
+        /// Creates a new view instance of type <see cref="CalendarDatePicker"/>.
+        /// </summary>
+        /// <param name="reactContext">The React context.</param>
+        /// <returns>The view instance.</returns>
+        protected override CalendarDatePicker CreateViewInstance(ThemedReactContext reactContext)
+        {
+            CalendarDatePicker picker = new CalendarDatePicker();
+            return picker;
+        }
+
+        /// <summary>
+        /// Binds the <see cref="CalendarDatePicker.DateChanged"/> event to a handler
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view instance.</param>
+        protected override void AddEventEmitters(ThemedReactContext reactContext, CalendarDatePicker view)
+        {
+            base.AddEventEmitters(reactContext, view);
+            view.DateChanged += OnDateChanged;
+        }
+
+        /// <summary>
+        /// Date changed update handler
+        /// </summary>
+        /// <param name="sender">The event sender.</param>
+        /// <param name="e">The event.</param>
+        private void OnDateChanged(object sender, CalendarDatePickerDateChangedEventArgs e)
+        {
+            if (!e.NewDate.HasValue)
+            {
+                return;
+            }
+
+            var datePicker = (CalendarDatePicker)sender;
+            DateTime? newDate;
+
+            newDate = e.NewDate.Value.DateTime;
+            
+            datePicker.GetReactContext().GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new ReactCalendarPickerEvent(datePicker.GetTag(), newDate));
+        }
+
+        /// <summary>
+        /// A CalendarDatePicker-specific event to communicate with JavaScript
+        /// </summary>
+        class ReactCalendarPickerEvent : Event
+        {
+            private readonly DateTime? _date;
+            /// <summary>
+            /// Creates an instance of the event
+            /// </summary>
+            /// <param name="viewTag">The viewtag of the instantiating view.</param>
+            /// <param name="date">Date to include in the event payload.</param>
+            public ReactCalendarPickerEvent(int viewTag, DateTime? date) :
+                base(viewTag)
+            {
+                _date = date;
+            }
+
+            /// <summary>
+            /// Internal name of the event
+            /// </summary>
+            public override string EventName
+            {
+                get
+                {
+                    return "topChange";
+                }
+            }
+
+            /// <summary>
+            /// Dispatches the event through the JavaScript bridge
+            /// </summary>
+            /// <param name="eventEmitter"></param>
+            public override void Dispatch(RCTEventEmitter eventEmitter)
+            {
+                JObject eventData;
+                if (_date.HasValue)
```

Comment:
We can consolidate this logic a bit with:
```
eventEmitter.receiveEvent(ViewTag, EventName, new JObject
{
    { "date", _date.HasValue ? _date.Value : null }
});
```

I assume the reason for the conditional here is that Newtonsoft.Json does not properly handle the nullable value?
```

I assume the reason for the conditional

Reply:
Ran a simple test:
```
            DateTime? nullDate = null;
            DateTime? date = DateTime.Now;

            Console.WriteLine(new JObject { { "date", nullDate } });
            Console.WriteLine(new JObject { { "date", date } });
            Console.WriteLine(new JObject { { "date", date.Value } });
```
And the output is:
```
{
  "date": null
}
{
  "date": "2018-07-10T15:19:10.6642496-04:00"
}
{
  "date": "2018-07-10T15:19:10.6642496-04:00"
}
```
So you can actually simplify the whole thing by removing the conditional altogether and just supplying the nullable `_date` value to the `JObject`.